frontend/packages/ceph-storage-plugin/src/components/dashboard-page/storage-dashboard/inventory.tsx

```diff
@@ -0,0 +1,117 @@
+import * as React from 'react';
+import * as _ from 'lodash';
+
+import { DashboardCard } from '@console/internal/components/dashboard/dashboard-card/card';
+import { DashboardCardBody } from '@console/internal/components/dashboard/dashboard-card/card-body';
+import { DashboardCardHeader } from '@console/internal/components/dashboard/dashboard-card/card-header';
+import { DashboardCardTitle } from '@console/internal/components/dashboard/dashboard-card/card-title';
+import { FirehoseResource } from '@console/internal/components/utils';
+import { InventoryItem } from '@console/internal/components/dashboard/inventory-card/inventory-item';
+import { K8sResourceKind } from '@console/internal/module/k8s';
+import {
+  NodeModel,
+  PersistentVolumeClaimModel,
+  PersistentVolumeModel,
+} from '@console/internal/models';
+import {
+  getNodeStatusGroups,
+  getPVCStatusGroups,
+  getPVStatusGroups,
+} from '@console/internal/components/dashboard/inventory-card/utils';
+import {
+  DashboardItemProps,
+  withDashboardResources,
+} from '@console/internal/components/dashboards-page/with-dashboard-resources';
+
+const cephStorageProvisiners = ['ceph.rook.io/block'];
+
+const getFilteredPVCsData = (pvcsData: K8sResourceKind[]): K8sResourceKind[] =>
+  pvcsData.filter((pvc: K8sResourceKind) =>
+    cephStorageProvisiners.includes(
+      _.get(pvc, 'metadata.annotations["volume.beta.kubernetes.io/storage-provisioner"]'),
+    ),
+  );
+
+const getFilteredPVsData = (pvsData: K8sResourceKind[]): K8sResourceKind[] =>
+  pvsData.filter((pv: K8sResourceKind) =>
+    cephStorageProvisiners.includes(
+      _.get(pv, 'metadata.annotations["pv.kubernetes.io/provisioned-by"]'),
+    ),
+  );
+
+const k8sResources: FirehoseResource[] = [
+  {
+    isList: true,
+    kind: PersistentVolumeModel.kind,
+    prop: 'pvs',
+  },
+  {
+    isList: true,
+    kind: NodeModel.kind,
+    prop: 'nodes',
+  },
+  {
+    isList: true,
+    kind: PersistentVolumeClaimModel.kind,
+    prop: 'pvcs',
+  },
+];
+
+const InventoryCard: React.FC<DashboardItemProps> = ({
+  watchK8sResource,
+  stopWatchK8sResource,
+  resources,
+}) => {
+  React.useEffect(() => {
+    k8sResources.forEach((r) => watchK8sResource(r));
+    return () => {
+      k8sResources.forEach((r) => stopWatchK8sResource(r));
+    };
+  }, [watchK8sResource, stopWatchK8sResource]);
+
+  const nodes = _.get(resources, 'nodes');
+  const nodesLoaded = _.get(nodes, 'loaded');
+  const nodesData = _.get(nodes, 'data', []) as K8sResourceKind[];
+
+  const pvcs = _.get(resources, 'pvcs');
```

Comment:
As @spadgett wrote, please avoid `_.get` on simple property access.

```ts
const pvcs = resources.pvcs;
```
is always better than
```ts
const pvcs = _.get(resources, 'pvcs');
```
since TypeScript cannot perform any validations when using `_.get` helper, which might lead to bugs (code compiles fine, blows up at runtime).

Reply:
Same as above, I'd suggest doing:

```ts
const pvcsLoaded = _.get(resources.pvcs, 'loaded');
const pvcsData = _.get(resources.pvcs, 'data', []) as K8sResourceKind[];
```

which trades some (but not too much) type checking for some convenience.

Reply:
@vojtechszocs ack, will keep that in mind.